[PVR][EPG] Feature: gridcontainer orientation

ksooo opened the pull request:
A feature that has been requested several times: PVR guide window with grid layout with channels as columns and timeline as rows.

![screenshot004](https://cloud.githubusercontent.com/assets/3226626/19785663/8e62395c-9c9a-11e6-88aa-0f94df42232a.png)

![screenshot003](https://cloud.githubusercontent.com/assets/3226626/19785458/b8dc379c-9c99-11e6-9d17-41e11d11cf11.png)

Please note that the skin changes are not intended to be merged with this PR (unless people say they want it in estuary). It's just a test/reference implementation. The respective commit will be removed from this PR before merge. 

@xhaggi for code review
@ronie heads-up for skinning engine extension

ksooo committed:
[epg] Skinning engine: epg grid: add support for alternative (horizontal) grid layout with channels as columns and timeline as rows.

ronie commented:
looks great! ;-)

xhaggi commented:
@ksooo we dropped this feature a year ago or so 🤔 
I wonder why it comes back now. Don't get me wrong I'm sure you re-added it in a more suitable way.
I'll take a look at it later.

ksooo commented:
As I wrote there are plenty of people requesting this feature. Are you sure we ever had this feature? Quick look through the commit history does not reveal anything like this.

xhaggi commented:
https://github.com/ksooo/xbmc/commit/a1fb9dcf4c2c94880422a1742667ebab84698790

phil65 commented:
Since we are in the process of removing all other viewtypes (now, next, channel) from the pvrguide window, perhaps we could give that window proper viewtype support just for the epg grid? That way skinners would have a sane way to optionally allow an additional layout.

ksooo commented:
Okay. Found it, was removed in mid of 2014. But I wonder why this happened. People consider this a neat feature and I agree (otherwise I never would have implemented it).

Maybe back then it was the right decision, because shit simply did not work? Now it does. ;-) I changed fundamentally things in this control during the last year. Maybe with this I "accidentally" fixed stuff that prevented the horizontal layout from working properly before.

ksooo commented:
@phil65 sorry, I have no idea what "proper view type support means".

ksooo commented:
... but would be happy to implement it.

phil65 commented:
Have a look at CGUIViewControl, that´s what would be needed for that window.

ksooo commented:
> Have a look at CGUIViewControl, that´s what would be needed for that window. That´s how mediawindows (and some other special cases) deal with viewtypes.

 The CGUIWindoPVRGuide is derived from CGUIMediaWindow. Does this mean it already supports "proper viewtypes"?

phil65 commented:
Possible. @xhaggi can probably tell us how feasible it is to implement that.

xhaggi commented:
@ksooo don't get me wrong again, if this is requested by several users and if it's done in a proper way I'm fine with it. Thanks for doing it 😄 

Regarding the view type support: The window supports it since I did the rewrite of the PVR views to be normal windows. You are able to specify a new views like you can do on every other media window e.g. "Vertical", "Horizontal".

xhaggi commented:
Ahh sorry but I'm not sure if it works currently because of the static view types "Now", "Next", "Channel" we want to drop for v18. After that it definitely should.

ksooo commented:
> don't get me wrong again, if this is requested by several users and if it's done in a proper way I'm fine with it. 

 yeah, this is how i got it in the first place. all good.

ksooo commented:
> Ahh sorry but I'm not sure if it works currently because of the static view types "Now", "Next", "Channel" we want to drop for v18. After that it definitely should.

 good to know. thanks for clarifying. thus, "proper viewtypes" (TM)  should be no big deal for v18, right.

phil65 commented:
> Ahh sorry but I'm not sure if it works currently because of the static view types "Now", "Next", "Channel" we want to drop for v18. After that it definitely should.

Yeah that´s why I asked, viewtype handling was always bit special for this window. If everything works like a normal MediaWindow after removal of Next, Now etc, then all perfect. 👍

xhaggi commented:
no, I'll create a PR which drops all other static view types except the timeline view and after that we're able to use the view type feature like on every other media window.

ksooo commented:
Rebased and removed skin reference  implementation which now can be found here: https://github.com/ksooo/xbmc/commit/860acc82599724b53a06fc6a2753a215c3f8a32e

jenkins build this please
